File: fontes/frontend/fenix_provider/lib/src/model/nfe/nfe_detalhe_imposto_icms_ufdest.dart
/*
Title: T2Ti ERP 3.0                                                                
Description: Model relacionado à tabela [NFE_DETALHE_IMPOSTO_ICMS_UFDEST] 
                                                                                
The MIT License                                                                 
                                                                                
Copyright: Copyright (C) 2021 T2Ti.COM                                          
                                                                                
Permission is hereby granted, free of charge, to any person                     
obtaining a copy of this software and associated documentation                  
files (the "Software"), to deal in the Software without                         
restriction, including without limitation the rights to use,                    
copy, modify, merge, publish, distribute, sublicense, and/or sell               
copies of the Software, and to permit persons to whom the                       
Software is furnished to do so, subject to the following                        
conditions:                                                                     
                                                                                
The above copyright notice and this permission notice shall be                  
included in all copies or substantial portions of the Software.                 
                                                                                
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,                 
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES                 
OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND                        
NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT                     
HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,                    
WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING                    
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR                   
OTHER DEALINGS IN THE SOFTWARE.                                                 
                                                                                
       The author may be contacted at:                                          
           t2ti.com@gmail.com                                                   
                                                                                
@author Albert Eije (alberteije@gmail.com)                    
@version 1.0.0
*******************************************************************************/
import 'dart:convert';


class NfeDetalheImpostoIcmsUfdest {
	int? id;
	int? idNfeDetalhe;
	double? valorBcIcmsUfDestino;
	double? valorBcFcpUfDestino;
	double? percentualFcpUfDestino;
	double? aliquotaInternaUfDestino;
	double? aliquotaInteresdatualUfEnvolvidas;
	double? percentualProvisorioPartilhaIcms;
	double? valorIcmsFcpUfDestino;
	double? valorInterestadualUfDestino;
	double? valorInterestadualUfRemetente;

	NfeDetalheImpostoIcmsUfdest({
		id,
		idNfeDetalhe,
		valorBcIcmsUfDestino = 0.0,
		valorBcFcpUfDestino = 0.0,
		percentualFcpUfDestino = 0.0,
		aliquotaInternaUfDestino = 0.0,
		aliquotaInteresdatualUfEnvolvidas = 0.0,
		percentualProvisorioPartilhaIcms = 0.0,
		valorIcmsFcpUfDestino = 0.0,
		valorInterestadualUfDestino = 0.0,
		valorInterestadualUfRemetente = 0.0,
	});

	static List<String> campos = <String>[
		'ID', 
		'VALOR_BC_ICMS_UF_DESTINO', 
		'VALOR_BC_FCP_UF_DESTINO', 
		'PERCENTUAL_FCP_UF_DESTINO', 
		'ALIQUOTA_INTERNA_UF_DESTINO', 
		'ALIQUOTA_INTERESDATUAL_UF_ENVOLVIDAS', 
		'PERCENTUAL_PROVISORIO_PARTILHA_ICMS', 
		'VALOR_ICMS_FCP_UF_DESTINO', 
		'VALOR_INTERESTADUAL_UF_DESTINO', 
		'VALOR_INTERESTADUAL_UF_REMETENTE', 
	];
	
	static List<String> colunas = <String>[
		'Id', 
		'Valor BC ICMS UF Destino', 
		'Valor BC FCP UF Destino', 
		'Percentual FCP UF Destino', 
		'Alíquota Interna UF Destino', 
		'Alíquota Interestadual UF Envolvidas', 
		'Percentual Provisório Partilha ICMS', 
		'Valor ICMS FCP UF Destino', 
		'Valor Interestadual UF Destino', 
		'Valor Interestadual UF Remetente', 
	];

	NfeDetalheImpostoIcmsUfdest.fromJson(Map<String, dynamic> jsonDados) {
		id = jsonDados['id'];
		idNfeDetalhe = jsonDados['idNfeDetalhe'];
		valorBcIcmsUfDestino = jsonDados['valorBcIcmsUfDestino']?.toDouble();
		valorBcFcpUfDestino = jsonDados['valorBcFcpUfDestino']?.toDouble();
		percentualFcpUfDestino = jsonDados['percentualFcpUfDestino']?.toDouble();
		aliquotaInternaUfDestino = jsonDados['aliquotaInternaUfDestino']?.toDouble();
		aliquotaInteresdatualUfEnvolvidas = jsonDados['aliquotaInteresdatualUfEnvolvidas']?.toDouble();
		percentualProvisorioPartilhaIcms = jsonDados['percentualProvisorioPartilhaIcms']?.toDouble();
		valorIcmsFcpUfDestino = jsonDados['valorIcmsFcpUfDestino']?.toDouble();
		valorInterestadualUfDestino = jsonDados['valorInterestadualUfDestino']?.toDouble();
		valorInterestadualUfRemetente = jsonDados['valorInterestadualUfRemetente']?.toDouble();
	}

	Map<String, dynamic> get toJson {
		Map<String, dynamic> jsonDados = <String, dynamic>{};

		jsonDados['id'] = id ?? 0;
		jsonDados['idNfeDetalhe'] = idNfeDetalhe ?? 0;
		jsonDados['valorBcIcmsUfDestino'] = valorBcIcmsUfDestino;
		jsonDados['valorBcFcpUfDestino'] = valorBcFcpUfDestino;
		jsonDados['percentualFcpUfDestino'] = percentualFcpUfDestino;
		jsonDados['aliquotaInternaUfDestino'] = aliquotaInternaUfDestino;
		jsonDados['aliquotaInteresdatualUfEnvolvidas'] = aliquotaInteresdatualUfEnvolvidas;
		jsonDados['percentualProvisorioPartilhaIcms'] = percentualProvisorioPartilhaIcms;
		jsonDados['valorIcmsFcpUfDestino'] = valorIcmsFcpUfDestino;
		jsonDados['valorInterestadualUfDestino'] = valorInterestadualUfDestino;
		jsonDados['valorInterestadualUfRemetente'] = valorInterestadualUfRemetente;
	
		return jsonDados;
	}
	

	String objetoEncodeJson() {
	  final jsonDados = toJson;
	  return json.encode(jsonDados);
	}
	
}

File: fontes/backend/php/fenix/src/model/nfe/NfeDetalheImpostoIcmsUfdest.php
<?php
/*******************************************************************************
Title: T2Ti ERP 3.0
Description: Model relacionado à tabela [NFE_DETALHE_IMPOSTO_ICMS_UFDEST] 
                                                                                
The MIT License                                                                 
                                                                                
Copyright: Copyright (C) 2021 T2Ti.COM                                          
                                                                                
Permission is hereby granted, free of charge, to any person                     
obtaining a copy of this software and associated documentation                  
files (the "Software"), to deal in the Software without                         
restriction, including without limitation the rights to use,                    
copy, modify, merge, publish, distribute, sublicense, and/or sell               
copies of the Software, and to permit persons to whom the                       
Software is furnished to do so, subject to the following                        
conditions:                                                                     
                                                                                
The above copyright notice and this permission notice shall be                  
included in all copies or substantial portions of the Software.                 
                                                                                
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,                 
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES                 
OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND                        
NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT                     
HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,                    
WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING                    
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR                   
OTHER DEALINGS IN THE SOFTWARE.                                                 
                                                                                
       The author may be contacted at:                                          
           t2ti.com@gmail.com                                                   
                                                                                
@author Albert Eije (alberteije@gmail.com)                    
@version 1.0.0
*******************************************************************************/
declare(strict_types=1);

class NfeDetalheImpostoIcmsUfdest extends EloquentModel implements \JsonSerializable
{
    /**
     * The table associated with the model.
     *
     * @var string
     */
    protected $table = 'NFE_DETALHE_IMPOSTO_ICMS_UFDEST';

    /**
     * Eager Loading - Relacionamentos que devem ser sempre carregados por padrão
     *
     * @var array
     */
	
    /**
     * Relations
     */
    public function nfeDetalhe()
    {
    	return $this->belongsTo(NfeDetalhe::class, 'ID_NFE_DETALHE', 'ID');
    }


    /**
     * Gets e Sets
     */
    public function getIdAttribute() 
	{
		return $this->attributes['ID'];
	}

	public function setIdAttribute($id) 
	{
		$this->attributes['ID'] = $id;
	}

    public function getValorBcIcmsUfDestinoAttribute() 
	{
		return (double)$this->attributes['VALOR_BC_ICMS_UF_DESTINO'];
	}

	public function setValorBcIcmsUfDestinoAttribute($valorBcIcmsUfDestino) 
	{
		$this->attributes['VALOR_BC_ICMS_UF_DESTINO'] = $valorBcIcmsUfDestino;
	}

    public function getValorBcFcpUfDestinoAttribute() 
	{
		return (double)$this->attributes['VALOR_BC_FCP_UF_DESTINO'];
	}

	public function setValorBcFcpUfDestinoAttribute($valorBcFcpUfDestino) 
	{
		$this->attributes['VALOR_BC_FCP_UF_DESTINO'] = $valorBcFcpUfDestino;
	}

    public function getPercentualFcpUfDestinoAttribute() 
	{
		return (double)$this->attributes['PERCENTUAL_FCP_UF_DESTINO'];
	}

	public function setPercentualFcpUfDestinoAttribute($percentualFcpUfDestino) 
	{
		$this->attributes['PERCENTUAL_FCP_UF_DESTINO'] = $percentualFcpUfDestino;
	}

    public function getAliquotaInternaUfDestinoAttribute() 
	{
		return (double)$this->attributes['ALIQUOTA_INTERNA_UF_DESTINO'];
	}

	public function setAliquotaInternaUfDestinoAttribute($aliquotaInternaUfDestino) 
	{
		$this->attributes['ALIQUOTA_INTERNA_UF_DESTINO'] = $aliquotaInternaUfDestino;
	}

    public function getAliquotaInteresdatualUfEnvolvidasAttribute() 
	{
		return (double)$this->attributes['ALIQUOTA_INTERESDATUAL_UF_ENVOLVIDAS'];
	}

	public function setAliquotaInteresdatualUfEnvolvidasAttribute($aliquotaInteresdatualUfEnvolvidas) 
	{
		$this->attributes['ALIQUOTA_INTERESDATUAL_UF_ENVOLVIDAS'] = $aliquotaInteresdatualUfEnvolvidas;
	}

    public function getPercentualProvisorioPartilhaIcmsAttribute() 
	{
		return (double)$this->attributes['PERCENTUAL_PROVISORIO_PARTILHA_ICMS'];
	}

	public function setPercentualProvisorioPartilhaIcmsAttribute($percentualProvisorioPartilhaIcms) 
	{
		$this->attributes['PERCENTUAL_PROVISORIO_PARTILHA_ICMS'] = $percentualProvisorioPartilhaIcms;
	}

    public function getValorIcmsFcpUfDestinoAttribute() 
	{
		return (double)$this->attributes['VALOR_ICMS_FCP_UF_DESTINO'];
	}

	public function setValorIcmsFcpUfDestinoAttribute($valorIcmsFcpUfDestino) 
	{
		$this->attributes['VALOR_ICMS_FCP_UF_DESTINO'] = $valorIcmsFcpUfDestino;
	}

    public function getValorInterestadualUfDestinoAttribute() 
	{
		return (double)$this->attributes['VALOR_INTERESTADUAL_UF_DESTINO'];
	}

	public function setValorInterestadualUfDestinoAttribute($valorInterestadualUfDestino) 
	{
		$this->attributes['VALOR_INTERESTADUAL_UF_DESTINO'] = $valorInterestadualUfDestino;
	}

    public function getValorInterestadualUfRemetenteAttribute() 
	{
		return (double)$this->attributes['VALOR_INTERESTADUAL_UF_REMETENTE'];
	}

	public function setValorInterestadualUfRemetenteAttribute($valorInterestadualUfRemetente) 
	{
		$this->attributes['VALOR_INTERESTADUAL_UF_REMETENTE'] = $valorInterestadualUfRemetente;
	}


    /**
     * Mapping
     */
    public function mapear($objeto)
    {
        if (isset($objeto)) {
            isset($objeto->id) ? $this->setIdAttribute($objeto->id) : $this->setIdAttribute(null);

			$this->setValorBcIcmsUfDestinoAttribute($objeto->valorBcIcmsUfDestino);
			$this->setValorBcFcpUfDestinoAttribute($objeto->valorBcFcpUfDestino);
			$this->setPercentualFcpUfDestinoAttribute($objeto->percentualFcpUfDestino);
			$this->setAliquotaInternaUfDestinoAttribute($objeto->aliquotaInternaUfDestino);
			$this->setAliquotaInteresdatualUfEnvolvidasAttribute($objeto->aliquotaInteresdatualUfEnvolvidas);
			$this->setPercentualProvisorioPartilhaIcmsAttribute($objeto->percentualProvisorioPartilhaIcms);
			$this->setValorIcmsFcpUfDestinoAttribute($objeto->valorIcmsFcpUfDestino);
			$this->setValorInterestadualUfDestinoAttribute($objeto->valorInterestadualUfDestino);
			$this->setValorInterestadualUfRemetenteAttribute($objeto->valorInterestadualUfRemetente);

			// vincular objetos
        }
    }

    /**
     * Serialization
     * {@inheritdoc}
     */
    public function jsonSerialize()
    {
        return [
			'id' => $this->getIdAttribute(),
			'valorBcIcmsUfDestino' => $this->getValorBcIcmsUfDestinoAttribute(),
			'valorBcFcpUfDestino' => $this->getValorBcFcpUfDestinoAttribute(),
			'percentualFcpUfDestino' => $this->getPercentualFcpUfDestinoAttribute(),
			'aliquotaInternaUfDestino' => $this->getAliquotaInternaUfDestinoAttribute(),
			'aliquotaInteresdatualUfEnvolvidas' => $this->getAliquotaInteresdatualUfEnvolvidasAttribute(),
			'percentualProvisorioPartilhaIcms' => $this->getPercentualProvisorioPartilhaIcmsAttribute(),
			'valorIcmsFcpUfDestino' => $this->getValorIcmsFcpUfDestinoAttribute(),
			'valorInterestadualUfDestino' => $this->getValorInterestadualUfDestinoAttribute(),
			'valorInterestadualUfRemetente' => $this->getValorInterestadualUfRemetenteAttribute(),
        ];
    }
}